tests/sources/test_cppcheck.py

```diff
@@ -101,57 +167,35 @@ def setUp(self):
             'projects/vehicles/controllers/ros_automobile/include'
         ]
-
-    def test_cppcheck_is_correctly_installed(self):
-        """Test Cppcheck is correctly installed."""
-        self.assertTrue(
-            find_executable(self.cppcheck) is not None,
-            msg='Cppcheck is not installed on this computer.'
-        )
-
-    def run_cppcheck(self, command):
-        """Run Cppcheck command and check for errors."""
-        if os.path.isfile(self.reportFilename):
-            os.remove(self.reportFilename)
-
-        os.system(command)
-
-        if os.path.isfile(self.reportFilename):
-            reportFile = open(self.reportFilename, 'r')
-            reportText = reportFile.read()
-            self.assertTrue(
-                len(reportText) == 0,
-                msg='Cppcheck detected some errors:\n\n%s' % reportText
-            )
-            reportFile.close()
-            os.remove(self.reportFilename)
-
-    def test_sources_with_cppcheck(self):
-        """Test Webots with Cppcheck."""
-        command = self.cppcheck + ' --enable=warning,style,performance,portability --inconclusive --force -q'
-        command += ' -j %s' % str(multiprocessing.cpu_count())
-        command += ' --inline-suppr --suppress=invalidPointerCast --suppress=useStlAlgorithm --suppress=uninitMemberVar '
-        command += ' --suppress=noCopyConstructor  --suppress=noOperatorEq'
-        # command += ' --xml '  # Uncomment this line to get more information on the errors
-        command += ' --output-file=' + self.reportFilename
-        for include in self.includeDirs:
-            command += ' -I\"' + os.path.normpath(self.WEBOTS_HOME + '/' + include) + '\"'
-        for source in self.skippedDirs:
-            command += ' -i\"' + os.path.normpath(self.WEBOTS_HOME + '/' + source) + '\"'
-        for source in self.sourceDirs:
-            command += ' \"' + os.path.normpath(self.WEBOTS_HOME + '/' + source) + '\"'
-        self.run_cppcheck(command)
-
-    def test_projects_with_cppcheck(self):
-        """Test projects with Cppcheck."""
+        files_diff = os.path.join(self.WEBOTS_HOME, 'tests', 'sources', 'files_diff.txt')
         command = self.cppcheck + ' --enable=warning,style,performance,portability --inconclusive --force -q '
         command += '--inline-suppr --suppress=invalidPointerCast --suppress=useStlAlgorithm -UKROS_COMPILATION '
         # command += '--xml '  # Uncomment this line to get more information on the errors
-        command += '--std=c++03 --output-file=' + self.reportFilename
-        for source in self.projectsSkippedDirs:
-            command += ' -i\"' + os.path.normpath(self.WEBOTS_HOME + '/' + source) + '\"'
-        for source in self.projectsSourceDirs:
-            command += ' \"' + os.path.normpath(self.WEBOTS_HOME + '/' + source) + '\"'
+        command += '--std=c++03 --output-file=\"' + self.reportFilename + '\"'
+        if os.path.isfile(files_diff):
```

Comment:
The content of this if is almost duplicated from the on of the `test_sources_with_cppcheck` function, wouldn't it be possible to make a function to avoid duplication?